DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewRecorder.swift

```diff
@@ -7,6 +7,33 @@
 #if os(iOS)
 import UIKit
 
+@_spi(Internal)
+public struct UIImageResource: Resource {
+    let image: UIImage
+    let tintColor: UIColor?
+
+    init(image: UIImage, tintColor: UIColor?) {
+        self.image = image
+        self.tintColor = tintColor
+    }
+
+    public var identifier: String {
+        var identifier = image.srIdentifier
+        if let tintColorIdentifier = tintColor?.srIdentifier {
+            identifier += tintColorIdentifier
+        }
+        return identifier
+    }
+
+    public var data: Data {
+        var image = self.image
+        if #available(iOS 13.0, *), let tintColor = tintColor {
+            image = image.withTintColor(tintColor)
+        }
+        return image.scaledDownToApproximateSize(512.KB)
```

Comment:
512 magic number? 

Should this be based on the resolution or actual size on the disk?

Reply:
I'll add a comment explaining the limitation. In short it's dictated by the intake limits.